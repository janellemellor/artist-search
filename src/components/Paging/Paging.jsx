import React from 'react';
import PropTypes from 'prop-types';

const Paging = ({ handlePageChange, page, artistResults }) => (
  <section> 
    <button onClick={() => handlePageChange(-25)} disabled={page === 0}>&lt;</button>
    <button onClick={() => handlePageChange(25)} disabled={artistResults.length < 25}>&gt;</button>
  </section>  
);

Paging.propTypes = {
  handlePageChange: PropTypes.func.isRequired,
  page: PropTypes.number.isRequired,
  artistResults: PropTypes.array.isRequired
};

export default Paging;


