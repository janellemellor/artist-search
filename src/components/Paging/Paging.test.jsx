import React from 'react';
import { shallow } from 'enzyme';
import Paging from '../Paging/Paging.jsx';

describe('Paging component', () => {
  it('matches the Paging snapshot', () => {
    const wrapper = shallow(<Paging handlePageChange={() => {}}artistResults="U2" page={1} />);
    expect(wrapper).toMatchSnapshot();
  });
});
