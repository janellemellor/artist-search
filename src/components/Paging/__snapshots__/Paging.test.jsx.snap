// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paging component matches the Paging snapshot 1`] = `
<section>
  <button
    disabled={false}
    onClick={[Function]}
  >
    &lt;
  </button>
  <button
    disabled={true}
    onClick={[Function]}
  >
    &gt;
  </button>
</section>
`;
